test/omega/test.lean

```diff
@@ -382,4 +382,13 @@ example (i j : Nat) (p : i ≥ j) : True := by
 
 example (i : Fin 7) : (i : Nat) < 8 := by omega
 
+-- Check that we correctly process base^(e+1) for constant base.
+example (x e : Nat) (hx : x < 2^(e+1)) : x < 2^e *2 := by omega
+
+-- Check that we correctly handle `e.succ`
+example (x e : Nat) (hx : x < 2^(e.succ)) : x < 2^e *2 := by omega
+
+-- Check that this works for integer base.
+example (x : Int) (e : Nat) (hx : x < 2^(e+1)) : x < 2^e *2 := by omega
```

Comment:
```suggestion
example (x : Int) (e : Nat) (hx : x < (2 : Int)^(e+1)) : x < 2^e *2 := by omega
```
Probably this is equivalent, but by putting the cast explicitly, we know for sure it's not doing the cast to `Int` after doing the exponentiation with a `Nat` base